### PR TITLE
cache some txs

### PR DESCRIPTION
**Pull request by Yrp**
**What does this PR do?**

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: src/main/java/org/tron/program/FullNode.java [hunk 1]
```diff
@@ -104,4 +125,46 @@ public static void shutdown(final Application app) {
...
+            e.printStackTrace();
+          } catch (BadItemException e) {
+            e.printStackTrace();
+          }
+        })));
+    ListenableFuture<?> future = Futures.allAsList(futures);
+    try {
+      future.get();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (hongz) [inline src/main/java/org/tron/program/FullNode.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1891&issues=AWf0QKUgrt4Kct_4mYKy&open=AWf0QKUgrt4Kct_4mYKy)

[ISSUE_KEY]: AWf0QKUgrt4Kct_4mYKy

File: src/main/java/org/tron/program/FullNode.java [hunk 2]
```diff
@@ -104,4 +125,46 @@ public static void shutdown(final Application app) {
...
+    } catch (ExecutionException e) {
+      e.printStackTrace();
```
Comment (hongz) [inline src/main/java/org/tron/program/FullNode.java]: Vulnerability Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1891&issues=AWf0QKUgrt4Kct_4mYKz&open=AWf0QKUgrt4Kct_4mYKz)

[ISSUE_KEY]: AWf0QKUgrt4Kct_4mYKz

File: src/main/java/org/tron/program/FullNode.java [hunk 3]
```diff
@@ -104,4 +125,46 @@ public static void shutdown(final Application app) {
...
+          } catch (ItemNotFoundException e) {
+            e.printStackTrace();
+          } catch (BadItemException e) {
+            e.printStackTrace();
+          }
+        })));
+    ListenableFuture<?> future = Futures.allAsList(futures);
+    try {
+      future.get();
+    } catch (InterruptedException e) {
```
Comment (hongz) [inline src/main/java/org/tron/program/FullNode.java]: Bug Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1891&issues=AWf0QKUgrt4Kct_4mYK0&open=AWf0QKUgrt4Kct_4mYK0)

[ISSUE_KEY]: AWf0QKUgrt4Kct_4mYK0

**Comment by olenheim** [comment]
#1860

**Pull request closed without merging**